syntax/include.php

```diff
@@ -136,6 +136,8 @@ function render($format, Doku_Renderer $renderer, $data) {
             if (in_array($id, $page_stack)) continue;
             array_push($page_stack, $id);
 
+            if (isset($flags['exclude']) && preg_match($flags['exclude'], $id)) continue;
```

Comment:
This must be before the `array_push($page_stack, $id);` above, otherwise pages are pushed onto the stack (that prevents endless recursive includes) but never popped. Further, `preg_match` triggers a warning when the regular expression is invalid. I think it is a good idea to use `@preg_match` instead, though this will make it harder for users to debug issues with their regular expressions. I don't know if there is any good way to get (and then display) a meaningful error message in this case.

Reply:
I'm a complete PHP noob and only just started using Dokuwiki, so your comment is much appreciated. I updated the PR!

> though this will make it harder for users to debug issues with their regular expressions

I think if we add a link to https://regex101.com/ to the documentation and document properly what is being matched on, that will be enough for people to debug issues with their regular expressions. In case of an invalid regular expression, nothing is excluded.